### DISCUSSION TITLE
how to switch to next tab

### DISCUSSION THREAD

**Discussion by Sherlock-Holo** [wezterm/wezterm#4790] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11, Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

GNOME

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

I am using wezterm with some tabs(sometimes the number of tabs may > 9), I want to know how to switch to the next tab or prev tab

```lua
local mykeys = {}
for i = 1, 9 do
  -- ALT + number to activate that tab
  table.insert(mykeys, {
    key = tostring(i),
    mods = 'ALT',
    action = act.ActivateTab(i - 1),
  })
end

-- move tab
table.insert(mykeys, { key = '{', mods = 'SHIFT|ALT', action = act.MoveTabRelative(-1) })
table.insert(mykeys, { key = '}', mods = 'SHIFT|ALT', action = act.MoveTabRelative(1) })
```

I already have some shortcut keys to move tabs, now I wonder how to switch the active tab, but not move the tab

**Comment by jankatins** [comment]
```lua
    -- Same as in Chrome
    -- mac
    {key="LeftArrow", mods="CMD|OPT", action=wezterm.action{ActivateTabRelative=-1}},
    {key="RightArrow", mods="CMD|OPT", action=wezterm.action{ActivateTabRelative=1}},
```

**Reply by Sherlock-Holo** [reply]
I try this

```lua
table.insert(mykeys, { key = '{', mods = 'CTRL|ALT', action = act.ActivateTabRelative(-1) })
table.insert(mykeys, { key = '}', mods = 'CTRL|ALT', action = act.ActivateTabRelative(1) })
```

but it doesn't work, did I make mistake?

**Reply by Sherlock-Holo** [reply, marked as answer]
ah I know the reason...

```lua
table.insert(mykeys, { key = '{', mods = 'SHIFT|ALT', action = act.MoveTabRelative(-1) })
table.insert(mykeys, { key = '}', mods = 'SHIFT|ALT', action = act.MoveTabRelative(1) })
```

use the `{` and `}`, because mods has the `SHIFT`, so when I press the `[` `]` buttons, exactly I press the `{` `}`

however when I set

```lua
table.insert(mykeys, { key = '{', mods = 'CTRL|ALT', action = act.ActivateTabRelative(-1) })
table.insert(mykeys, { key = '}', mods = 'CTRL|ALT', action = act.ActivateTabRelative(1) })
```

there is not `SHIFT` in the mods, so when I press the `[` `]` buttons, it just `[` `]` 

show the answer is, I should set

```lua
-- switch active tab
table.insert(mykeys, { key = '[', mods = 'CTRL|ALT', action = act.ActivateTabRelative(-1) })
table.insert(mykeys, { key = ']', mods = 'CTRL|ALT', action = act.ActivateTabRelative(1) })
```

**Comment by sunbearc22** [comment]
I discovered these WezTerm default keybindings:
- Use <kbd>Super</kbd><kbd>Shift</kbd><kbd>[</kbd> to go to the previous tab and <kbd>Super</kbd><kbd>Shift</kbd><kbd>]</kbd> for the next tab. 
- These can also be triggered with <kbd>Ctrl </kbd><kbd>Tab</kbd> (next tab) and <kbd>Ctrl</kbd><kbd>Shift</kbd><kbd>Tab</kbd>(previous tab).

So no programming is needed, unless you want to change them.
